csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -30,20 +30,25 @@ int64_t countNonTrivialIterDomains(const TensorView* tv) {
 // Note [ Allocation Order Mapping ]
 //
 // Map allocation domain from ref to target's logical domain to construct a new
-// allocation domain for target. The objective is to have target in a similar
-// memory format as with ref.
+// allocation domain for target. The objective is to set target with the similar
+// inner most dimensions in storage to facilitate larger vectorization factor.
 //
 // The propagation rule explained in an example, given inputs:
 //   ref's allocation domain
-//     {iS0[i0], ir1[i1], iS2[i2]}
+//     {iS9[i5], iS0[i0], iS1[i1], iS2[i2]}
 //   target's logical domain
-//     {iS3[i3], iS4[i4], ir5[i1], iS6[i5], iS7[i2], ir8[1]}
+//     {iS3[i3], iS4[i4], iS5[i1], iS6[i5], iS7[i2], ir8[1]}
 //
-// 1. we project iter domains from targets' logical domain which has an exact
-// map to ref's allocation domain. (sharp-edge 0: we exclude mapping from
-// iteration id on ref to reduction id on target to avoid unnecessary
-// re-ordering which exposes #2202).
-//   mapped_ids {ir5[i1], iS7[i2]}
+// 1. we project iter domains from refs' allocation domain to targets' logical
+// domains, starting from the fast iter domains, until we fail to find an exact
+// map for a non trivial iter domain, note: we would skip no mapping for trivial
+// iter domains in refs. (sharp-edge 0: we exclude mapping from iteration id on
+// ref to reduction id on target to avoid unnecessary re-ordering which exposes
+// #2202).
+//   we go through iS2[i2] (mapped)
+//              -> iS1[i1] (mapped)
+//              -> iS0[i0] (break, since there's no mapping)
```

Comment:
In the PR description, you said

> * Allocation order propagation is limited only to consecutive outermost IterDomains of reference tensor.

Did you mean innermost instead? That would match the example here. If outermost, we'd go through `i5` first.

Reply:
missed the PR description. Thanks for catching this one.